pkg/scanner/v1/trivy.go

```diff
@@ -61,6 +64,58 @@ func (t *TrivyScanner) ParseReport(reportData []byte) (map[string]interface{}, e
 	return jsonMap, nil
 }
 
+//func (t *TrivyScanner) GetExpectedFields() []string {
+//	return []string{"Results", "Vulnerabilities", "Severity", "PkgName"}
+//}
+
+func (t *TrivyScanner) GenerateFingerprint(image string, reportData []byte, celExpression string) (string, error) {
+	parsedReport, err := t.ParseReport(reportData)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse report: %w", err)
+	}
+
+	fmt.Printf("[DEBUG] Evaluating CEL: %s\n", celExpression)
+
+	// Use google.protobuf.Struct for CEL compatibility
+	env, err := cel.NewEnv(
+		cel.Declarations(
+			decls.NewVar("report", decls.NewObjectType("google.protobuf.Struct")),
```

Comment:
We can declare the type to be a custom type based on the scanner type in the future. Some exploration here: https://www.perplexity.ai/search/cel-declarations-decls-newvar-VzBu9QRpRg.yrAuYEBSXSg